data/src/test/java/org/akvo/flow/data/entity/FormHeaderParserTest.java

```diff
@@ -0,0 +1,113 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.List;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@SmallTest
+@PrepareForTest(TextUtils.class)
+@RunWith(PowerMockRunner.class)
+public class FormHeaderParserTest {
+
+    @Before
+    public void setUp() {
+        PowerMockito.mockStatic(TextUtils.class);
+        when(TextUtils.isEmpty(any(CharSequence.class))).thenAnswer(new Answer<Boolean>() {
+            @Override
+            public Boolean answer(InvocationOnMock invocation) {
+                CharSequence a = (CharSequence) invocation.getArguments()[0];
+                return !(a != null && a.length() > 0);
+            }
+        });
+    }
+
+    @Test
+    public void parseMultipleShouldReturnEmpyIfNullResponse() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        final List<ApiFormHeader> results = formHeaderParser.parseMultiple(null);
+
+        assertTrue(results.isEmpty());
+    }
+
+    @Test
+    public void parseMultipleShouldReturnEmpyIfEmptyResponse() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        final List<ApiFormHeader> results = formHeaderParser.parseMultiple("");
+
+        assertTrue(results.isEmpty());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void parseMultipleShouldThrowExceptionIfInvalidResponseLength() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        formHeaderParser.parseMultiple("abc,cde");
+    }
+
+    @Test
+    public void parseMultipleShouldParseCorrectResponse() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        final List<ApiFormHeader> results = formHeaderParser.parseMultiple(",1,cde,abc,cde,6.0,cde,true,33");
+
+        assertFalse(results.isEmpty());
+    }
+
+    @Test
+    public void parseOneShouldReturnCorrectValuesIfStartsWithComa() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        ApiFormHeader result = formHeaderParser.parseOne(",1,cde,abc,cde,6.0,cde,true,33");
+
+        assertNotNull(result);
+        assertEquals("33", result.getRegistrationSurveyId());
+    }
+
+    @Test
+    public void parseOneShouldReturnCorrectValuesIfStartsWithoutComa() {
+        FormHeaderParser formHeaderParser = new FormHeaderParser();
+
+        ApiFormHeader result = formHeaderParser.parseOne("1,cde,abc,cde,6.0,cde,true,33");
```

Comment:
This one I dont get very well.  So even if there is no comma at the beginning the parsing should still identify correctly the format? is it because the format is different for some forms? Are there cases with the comma where something comes before the comma that is not the `id`?

Reply:
yes the first item is either empty or it can be the phone number, not sure why it's like this.

Reply:
https://github.com/akvo/akvo-flow/blob/ffac2098b0e5b66471b65c28aaa3f62999531682/GAE/src/org/waterforpeople/mapping/app/web/SurveyManagerServlet.java#L141

Reply:
https://github.com/akvo/akvo-flow/blob/ffac2098b0e5b66471b65c28aaa3f62999531682/GAE/src/org/waterforpeople/mapping/app/web/SurveyManagerServlet.java#L216